openbb_platform/providers/alpha_vantage/openbb_alpha_vantage/models/equity_historical.py

```diff
@@ -180,5 +186,6 @@ def transform_data(
             {"date": date, **{extract_key_name(k): v for k, v in values.items()}}
             for date, values in data.items()
         ]
+        data = filter_by_dates(data, query.start_date, query.end_date)
```

Comment:
I think we should do the filtering in the `extract_data` function as technically it's more suitable over there. The `filter_by_dates` function would then require a little tweak to work for a `dict` type object.